This test is failing for me with:

```
TEST-START | test_sms_with_attachments.py TestSmsWithAttachments.test_sms_send
TEST-UNEXPECTED-ERROR | test_sms_with_attachments.py TestSmsWithAttachments.test_sms_send | NoSuchElementException: NoSuchElementException: Unable to locate element: .loading-screen


Traceback (most recent call last):
  File "/home/florinstrugariu/.virtualenvs/gaia/local/lib/python2.7/site-packages/marionette/marionette_test.py", line 268, in run
    testMethod()
  File "/home/florinstrugariu/gaia/tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_sms_with_attachments.py", line 33, in test_sms_send
    camera = activities_list.tap_camera()
  File "/home/florinstrugariu/gaia/tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py", line 57, in tap_camera
    camera.wait_for_loading_spinner_hidden()
  File "/home/florinstrugariu/gaia/tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py", line 47, in wait_for_loading_spinner_hidden
    loading_spinner = self.marionette.find_element(*self._loading_screen_locator)
  File "/home/florinstrugariu/.virtualenvs/gaia/local/lib/python2.7/site-packages/marionette/marionette.py", line 1354, in find_element
    response = self._send_message('findElement', 'value', **kwargs)
  File "/home/florinstrugariu/.virtualenvs/gaia/local/lib/python2.7/site-packages/marionette/decorators.py", line 36, in _
    return func(*args, **kwargs)
  File "/home/florinstrugariu/.virtualenvs/gaia/local/lib/python2.7/site-packages/marionette/marionette.py", line 634, in _send_message
    self._handle_error(response)
  File "/home/florinstrugariu/.virtualenvs/gaia/local/lib/python2.7/site-packages/marionette/marionette.py", line 673, in _handle_error
    raise errors.NoSuchElementException(message=message, status=status, stacktrace=stacktrace)
TEST-INFO took 123878ms

SUMMARY
-------
passed: 0
failed: 1
todo: 0

```

--- a/tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py
+++ b/tests/python/gaia-ui-tests/gaiatest/apps/camera/app.py
@@ -50,6 +50,14 @@ class Camera(Base):
         self.wait_for_capture_ready()
         self.wait_for_element_not_displayed(*self._loading_screen_locator)
 
+    def wait_for_loading_spinner_hidden(self):
+        loading_spinner = self.marionette.find_element(*self._loading_screen_locator)
+        Wait(self.marionette).until(expected.element_not_displayed(loading_spinner))
+
+    def wait_for_loading_spinner_displayed(self):
+        loading_spinner = self.marionette.find_element(*self._loading_screen_locator)
+        Wait(self.marionette).until(expected.element_displayed(loading_spinner))
+
     @property
     def camera_mode(self):
         return self.marionette.find_element(*self._controls_locator).get_attribute('data-mode')
@@ -85,7 +93,9 @@ class Camera(Base):
         self.wait_for_thumbnail_visible()
 
     def tap_capture(self):
-        self.marionette.find_element(*self._capture_button_locator).tap()
+        capture_button = self.marionette.find_element(*self._capture_button_locator)
+        Wait(self.marionette).until(expected.element_enabled(capture_button))
+        capture_button.tap()
 
     def tap_select_button(self):
         select = self.marionette.find_element(*self._select_button_locator)

--- a/tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py
+++ b/tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py
@@ -59,6 +59,8 @@ class Activities(Base):
         camera = Camera(self.marionette)
         self.wait_for_condition(lambda m: self.apps.displayed_app.name == camera.name)
         self.apps.switch_to_displayed_app()
+        camera.wait_for_loading_spinner_displayed()
+        camera.wait_for_loading_spinner_hidden()
         camera.wait_for_capture_ready()
         return camera
 

